### PR TITLE
Update ONNX Export for Interpolate in Opset 11

### PR DESCRIPTION
**Pull request by lara-hdr**
- Add support for linear and cubic interpolate in opset 11.
- Add support for 1d and 3d interpolate in nearest mode for opset 7 and 8.
- Add tests for all cases of interpolate in ORT tests (nearest/linear/cubic, 1d/2d/3d, upsample/downsample).

### CODE REVIEW

**Comment by BowenBao** [comment]
Should we also add support for interpolate linear in opset9(and 7,8,10)? https://github.com/pytorch/pytorch/issues/22906

**Comment by lara-hdr** [comment]
> Should we also add support for interpolate linear in opset9(and 7,8,10)? #22906

We cannot support opset 7 to 10; ONNX's linear Resize does not match Pytorch's interpolation for opset 10 and below, but it was updated in opset 11 to match Pytorch's behavior (https://github.com/onnx/onnx/pull/2057).

**Comment by lara-hdr** [comment]
CI failing because of a bug in onnx:Resize that was fixed in v0.5.0.
Waiting on ORT version bump in https://github.com/pytorch/pytorch/pull/24414 to fix the CI failure.

**Review by dzhulgakov — COMMENTED** [review]
Looks reasonable, but I didn't review all details of interpolation logic - getting an extra pair of eyes would be useful.

Please fix the lint

**Review by BowenBao — COMMENTED** [review]
Thanks for bringing back linear interpolation :). Some inline comments

File: torch/onnx/symbolic_opset9.py [hunk 1]
```diff
@@ -719,31 +719,20 @@ def replication_pad(g, input, padding):
 
 def _interpolate(name, dim, interpolate_mode):
     def symbolic_fn(g, input, output_size, align_corners=None):
-        align_corners = sym_help._maybe_get_scalar(align_corners)
```
[the remark this answers is not in the document]
Comment (lara-hdr) [reply torch/onnx/symbolic_opset9.py]: this was not on purpose. thanks!

**Comment by houseroad** [comment]
Fix the conflict? Since I merged some others.

**Comment by lara-hdr** [comment]
@houseroad conflicts are fixed. 
thanks

File: torch/nn/functional.py [hunk 2]
```diff
@@ -2477,7 +2477,7 @@ def _output_size(dim):
 
         # make scale_factor a tensor in tracing so constant doesn't get baked in
         if torch._C._get_tracing_state():
-            return [(torch.floor((input.size(i + 2) * torch.tensor(float(scale_factors[i]))).float())) for i in range(dim)]
+            return [(torch.floor((input.size(i + 2).float() * torch.tensor(scale_factors[i], dtype=torch.float32)).float())) for i in range(dim)]
```
Comment (houseroad) [inline torch/nn/functional.py]: why explicitly convert float32?

**Review by houseroad — APPROVED** [review]
LGTM

**Pull request closed without merging**